src/adapters/methods/lora.py

```diff
@@ -429,9 +447,12 @@ def compose_single(self, adapter_setup: str, state: LoRAState, lvl: int = 0) ->
         return state._replace(hidden_states=hidden_states)
 
     def forward(self, input_states: torch.Tensor):
-        weight = torch.transpose(self.weight, -2, -1) if self.fan_in_fan_out else self.weight
-        # result shape: <batch_size> x <seq_len> x <head_dim>
-        layer_output = F.linear(input_states, weight, bias=self.bias)
+        if self.fan_in_fan_out:
+            weight = torch.transpose(self.weight, -2, -1) if self.fan_in_fan_out else self.weight
+            # result shape: <batch_size> x <seq_len> x <head_dim>
+            layer_output = F.linear(input_states, weight, bias=self.bias)
+        else:
+            layer_output = super().forward(input_states)
```

Comment:
Which forward method is called here since this does not inherit from nn.Linear anymore?

Reply:
the subclasses of this (`LoRALinearTorch`, `LoRALinear4bit`, `LoRALinear8bitLt`),  inherit from different types of linear layers